src/Agent.Plugins/PipelineCache/SavePipelineCacheV0.cs

```diff
@@ -29,6 +29,19 @@ public class SavePipelineCacheV0 : PipelineCacheTaskPluginBase
             string salt,
             CancellationToken token)
         {
+            
+            if(!context.Variables.ContainsKey("agent.jobstatus"))
+            {
+                return;
+            }
+
+            var JobStatus = context.Variables["agent.jobstatus"];
+            if(JobStatus != null && !JobStatus.Value.Equals(Microsoft.TeamFoundation.Build.WebApi.TaskResult.Succeeded.ToString()))
```

Comment:
I am not sure if this is the best way to check the job status, to avoid HardCoding.